integration_tests/run_pyspark_from_build.sh

```diff
@@ -41,13 +41,17 @@ else
     RUN_DIR="$SCRIPTPATH"/target/run_dir
     mkdir -p "$RUN_DIR"
     cd "$RUN_DIR"
-    "$SPARK_HOME"/bin/spark-submit --jars "${ALL_JARS// /,}" \
-        --conf "spark.driver.extraJavaOptions=-Duser.timezone=GMT $COVERAGE_SUBMIT_FLAGS" \
-        --conf 'spark.executor.extraJavaOptions=-Duser.timezone=GMT' \
-        --conf 'spark.sql.session.timeZone=UTC' \
-        --conf 'spark.sql.shuffle.partitions=12' \
-        $SPARK_SUBMIT_FLAGS \
-        "$SCRIPTPATH"/runtests.py --rootdir "$SCRIPTPATH" "$SCRIPTPATH"/src/main/python \
+    export EXTRA_CP="${ALL_JARS// /:}"
+    #"$SPARK_HOME"/bin/spark-submit --jars "${ALL_JARS// /,}" \
+    #    --conf "spark.driver.extraJavaOptions=-Duser.timezone=GMT $COVERAGE_SUBMIT_FLAGS" \
+    #    --conf 'spark.executor.extraJavaOptions=-Duser.timezone=GMT' \
+    #    --conf 'spark.sql.session.timeZone=UTC' \
+    #    --conf 'spark.sql.shuffle.partitions=12' \
+    #    $SPARK_SUBMIT_FLAGS \
+
+    RUN_PARALELLE=${RUN_PARALELLE:-4}
```

Comment:
could try w/ `nvidia-smi --query-gpu=memory.total --format=csv,noheader` to decide the parallelism

Reply:
I am fine with that as a default, but I would like to have us have something through maven that we can use to either override the parallelism or to set the target amount of memory each level of parallelism will need.

Reply:
@revans2  
To run on the local maven build, we can use `export RUN_PARALELLE=xxx && mvn package'

For the pre-merge pipeline, it is not easy to dynamically change the Jenkins parameter `RUN_PARALLEL`, because the pre-merge pipeline is auto-triggered by the PRs, we have no change to input this parameter. 

For the nightly/integration pipelines, we can manually change the `RUN_PARALLEL` parameter to specify the parallelisms and manually trigger the pipelines